src/Orleans.Autofac/AutofacOrleansDependencyScope.cs

```diff
@@ -0,0 +1,123 @@
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Security;
+using System.Linq;
+using Autofac;
+using Orleans.Runtime;
+
+namespace Orleans.Autofac
+{
+    /// <summary>
+    /// Autofac implementation of the <see cref="IDependencyScope"/> interface.
+    /// </summary>
+    [SecurityCritical]
+    public class AutofacOrleansDependencyScope : IDependencyScope
+    {
+        private bool _disposed;
+
+        readonly ILifetimeScope _lifetimeScope;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AutofacOrleansDependencyScope"/> class.
+        /// </summary>
+        /// <param name="lifetimeScope">The lifetime scope to resolve services from.</param>
+        public AutofacOrleansDependencyScope(ILifetimeScope lifetimeScope)
+        {
+            if (lifetimeScope == null) throw new ArgumentNullException("lifetimeScope");
+
+            _lifetimeScope = lifetimeScope;
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the <see cref="AutofacOrleansDependencyScope"/> class.
+        /// </summary>
+        [SecuritySafeCritical]
+        ~AutofacOrleansDependencyScope()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Gets the lifetime scope for the current dependency scope.
+        /// </summary>
+        public ILifetimeScope LifetimeScope
+        {
+            get { return _lifetimeScope; }
+        }
+
+        /// <summary>
+        /// Try to get a service of the given type.
+        /// </summary>
+        /// <param name="serviceType">ControllerType of service to request.</param>
+        /// <returns>An instance of the service, or null if the service is not found.</returns>
+        [SecurityCritical]
+        public object GetService(Type serviceType)
+        {
+            return _lifetimeScope.ResolveOptional(serviceType);
+        }
+
+        /// <summary>
+        /// Try to get a list of services of the given type.
+        /// </summary>
+        /// <param name="serviceType">ControllerType of services to request.</param>
+        /// <returns>An enumeration (possibly empty) of the service.</returns>
+        [SecurityCritical]
+        public IEnumerable<object> GetServices(Type serviceType)
```

Comment:
Nope, just a leftover, the interface was cleaned up, the implementation was not. Sorry.